extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -312,10 +362,12 @@ public void run()
               public ConsumerRecords<byte[], byte[]> call() throws Exception
               {
                 try {
+                  status = Status.READING;
                   return consumer.poll(POLL_TIMEOUT);
                 }
-                finally {
-                  startedReading = true;
+                catch (OffsetOutOfRangeException e) {
+                  status = Status.WAITING;
```

Comment:
I removed the waiting status and handled interrupts better. The diff got a lot messier but I basically just wrapped the main `while (stillReading)` loop into a try/catch so that even if we are interrupted we can still persist to disk before shutting down